### PR TITLE
Bout detection improvements and simplification

### LINKED ISSUE #606: g.getbout probably misclassifies bouted time as unbouted
When using `part5_agg2_60seconds = TRUE` and `boutdur.mvpa = c(1, 5, 10)`, I was expecting not to get any time into the **dur_day_MOD_unbt** and **dur_day_VIG_unbt** categories (the shortest bout lenght is equal to the epoch length, so any epoch above the moderate threshold should be classified as a bout of 1 minute). However, I get some minutes in the unbouted category, and I have noticed that all the **dur_day_MVPA_bts_1_5_min** have actually a duration of at least 2 min (and not 1 min as expected). I think it is related to the piece of code looking at the first and last minute of every bout meeting the bout.criter (g.getbout, bout.metric = 6).

### PR DESCRIPTION
**Pull request by vincentvanhees**
@jhmigueles this PR integrates your revisions to the algorithm from PR #709, I copied your exact code without making modifications.

Further, I made the following complementary changes to other parts of GGIR:
- Arguments bout.metric and closedbout deprecated, as well as the code for all past bout metrics. I have removed them from both documentation (GGIR.Rd and g.getbout.Rd, and GGIR.Rmd) and from various functions.
- Unit test rewritten to focus on all possible scenarios where it goes wrong and no longer using random factors because that makes it impossible to trace back why a unit-test fails.
- Some other unit tests where specific to older bout.metrics which I have now updated to be specific for the latest version.
- Minor improvements to warning and stop calls related to myfun.
- Removed a lot of spaces from the main vignette tables on output variables. I do not know how they arrived there, but these make it very hard to read the file. Maybe they originated by copying the code from a text editor such as MS Word?

Fixes #606 

Checklist before merging:

- [x] Existing tests still work (check by running the test suite, e.g. from RStudio).
- [x] Added tests (if you added functionality) or fixed existing test (if you fixed a bug).
- [x] Updated or expanded the documentation.
- [x] Updated release notes in `inst/NEWS.Rd` with a user-readable summary. Please, include references to relevant issues or PR discussions.
- [ ] Added your name to the contributors lists in the `DESCRIPTION` and `CITATION.cff` files.

### CODE REVIEW

File: R/g.analyse.perday.R [hunk 1]
```diff
@@ -521,29 +521,23 @@ g.analyse.perday = function(ndays, firstmidnighti, time, nfeatures,
...
                       p = which(varnum * UnitReScale >= params_phyact[["mvpathreshold"]][mvpai]); rr1[p] = 1
                       getboutout = g.getbout(x = rr1, boutduration = boutduration,
-                                             boutcriter = params_phyact[["boutcriter"]],
-                                             closedbout = params_phyact[["closedbout"]],
-                                             bout.metric = params_phyact[["bout.metric"]], ws3 = ws3)
-                      mvpa[5] = length(which(getboutout$x == 1))   / (60/ws3) #time spent MVPA in minutes
+                                             boutcriter = params_phyact[["boutcriter"]], ws3 = ws3)
+                      mvpa[5] = length(which(getboutout == 1))   / (60/ws3) #time spent MVPA in minutes
                       # METHOD 6: time spent above threshold 10 minutes
                       boutduration = params_phyact[["mvpadur"]][3] * (60/ws3) # per ten minutes
```
Comment (jhmigueles) [inline R/g.analyse.perday.R:535]: Just realized that GGIR expects that `mvpadur` is of length = 3. Should we say this in the documentation? Or do a for look through the `mvpadur` defined by the user to make it more flexible as in part 5?

Comment (vincentvanhees) [reply R/g.analyse.perday.R:535]: You do not have to do this for me, but if you want to work on it then feel free to do so. Ideally, as a separate issue and PR as it does not relate to bout detection directly.

File: tests/testthat/test_ggetbout.R [hunk 2]
```diff
@@ -1,91 +1,54 @@
...
+    
+    
+    # Metric 7 - bout length epquals epoch length
+    boutduration = 1
+    xtest = c(rep(0, 10 * N), 1, rep(0, 10 * N))
+    bouts1 = g.getbout(x = xtest, boutduration = boutduration, boutcriter = boutcriter, ws3 = ws3)
+    expect_equal(sum(bouts1), 1)
+    expect_equal(bouts1, xtest)
+    expect_equal(bouts1, xtest)
+  }
```
Comment (jhmigueles) [inline tests/testthat/test_ggetbout.R:49]: Should we also test that the bout starts/ends at the epochs it is expected? We then would make sure that light PA bouts do not overlap previously detected MVPA or Inactivity bouts. I can add this to the test if you agree.

**Review by jhmigueles — APPROVED** [review]
I have added the check on the start and end epoch of each bout to test_ggetbout. The rest of changes look good to me.

**Pull request merged**